frontend/src/metabase/visualizations/components/TableInteractive.jsx

```diff
@@ -814,27 +888,28 @@ export default class TableInteractive extends Component {
               <Grid
                 ref={ref => (this.header = ref)}
                 style={{
+                  position: "absolute",
                   top: 0,
                   left: 0,
-                  right: 0,
                   height: headerHeight,
-                  position: "absolute",
                   overflow: "hidden",
+                  width: width,
+                  paddingRight: getScrollBarSize(),
                 }}
                 className="TableInteractive-header scroll-hide-all"
                 width={width || 0}
                 height={headerHeight}
                 rowCount={1}
                 rowHeight={headerHeight}
-                // HACK: there might be a better way to do this, but add a phantom padding cell at the end to ensure scroll stays synced if main content scrollbars are visible
-                columnCount={cols.length + 1}
-                columnWidth={props =>
-                  props.index < cols.length ? this.getColumnWidth(props) : 50
-                }
+                columnCount={cols.length + gutterColumn}
+                columnWidth={this.getColumnWidth}
                 cellRenderer={props =>
-                  props.columnIndex < cols.length
-                    ? this.tableHeaderRenderer(props)
-                    : null
+                  gutterColumn && props.columnIndex === 0
+                    ? () => undefined // we need a phantom cell to properly offset columns
```

Comment:
This follows the pattern in the react-virtualized docs to insert empty cells to create space for the frozen row to sit:
https://bvaughn.github.io/react-virtualized/#/components/ScrollSync